accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -46,214 +46,205 @@ extension MyFancyList<T> on List<T> {
 
-Using such an extension as:
+You still cannot *export* two extensions with the same name. The rules for export makes it a compile-time error to add two declarations with the same name to the export scope of a library.
 
-```dart
-Object o = ...;
-String s = ...;
-print(s.equals(o));  // Compile-time type error.
-```
+##### Applicability
 
-will fail. While we could make it work by inferring `T` as `Object`, we don't. We infer `T` *only* based on the receiver type, and therefore `T` is `String`, and `o` is not a valid argument (at least not when we remove implicit downcasts).
+An extension `E` is *applicable* to a simple or composite member invocation with corresponding member *basename* *m* and target expression `e`, where `e` has static type *S*, if
 
-### Extension Conflict Resolution
+- The invocation is an *instance* member invocation. That is the case if the expression `e` does not denote a prefix or a class, mixin or extension declaration *(then the member invocation would be a static invocation)*, and it is not an explicit extension application. An instance member invocation on `e` will always begin by evaluating `e` to an object, and then continue by performing an instance member invocation on that object.
+- The type *S*  does not have a member with the basename  *m*. For this, the type `dynamic`is considered as having all member names, and an expression of type `Never` or `void` cannot occur as the target of a member invocation, so none of these can ever have applicable extensions. Function types and the type `Function` are considered as having a `call` member. *This ensure that if there is an applicable extension, the existing invocation would otherwise be a compile-time error*. Members of `Object` exists on all types, so they can never be the target of implicit member invocations _(they can also not be declared as extension members)_.
```

Comment:
This seems to imply that an extension setter `x=` will never be applicable if the `on` type has an `x` getter, right?  I think this is a change.  The analyzer and the CFE allow this.

Reply:
This is not a *change*, it has always been the intent.
The existing specification states:

> For any member access, `x.foo`, `x.bar()`, `x.baz = 42`, `x(42)`, `x[0] = 1` or `x + y`, including `null`-aware and cascade accesses which effectively desugar to one of those direct accesses, and including implicit member accesses on this, the language first checks whether the static type of `x` has a member with the same `base name` as the operation. That is, if it has a corresponding instance member, respectively, a `foo` method or getter or a `foo=` setter. a `bar `member or `bar=` setter, a `baz` member or `baz=` setter, a `call` method, a `[]=` operator or a `+` operator. If so, then the operation is unaffected by extensions.

Notice the use of "base name", and explicitly mentioning both `foo` and `foo=`.

If it is not allowed by tools, that is an error.

The only difference I might have made here is that I treat `[]` and `[]=` as having the same basename. (And I'm not entirely sure about that, but it's probably a good idea to treat `[]=` as a setter in as many ways as possible - every time we didn't, it has come back to bite us).

Reply:
The CFE allowed this as of 2 pm yesterday, are you sure dart2js and and the VM are correctly rejecting it?

Reply:
I think maybe you were misdiagnosing this? Maybe the CFE implemented extension method resolution at all at the time - and it was telling you that it didn't know anything about a setter named `x`?  In any case, in a build from last week, the CFE emitted no errors for the example you list above.